V3 changes

mattsaxon opened the pull request:
This PR has 2 commits in it. The first is to cover V3 firmware assuming that pysonofflan is upgraded to V3 on PyPi. The second supports deployment of this in a local directory (outside of PyPi).

Note it also contains previous PRs (on icons and removal of unecessary update calls) that have yet to be accepted into the main branch. https://github.com/beveradb/sonoff-lan-mode-homeassistant/pull/65

mattsaxon committed:
Support V3 firmware configurations

These changes add the config items that will work with V3 firmware. However the code will work with either V2 or V3 pysonofflan

mattsaxon committed:
Changes to support side by side V2 and V3

With the r3 version of pysonofflan stored in a subdirectory of the HA custom_components directory

mattsaxon committed:
Create custom_components.zip

A single file with the component and side by side dependency of pysonofflan (v3)

mattsaxon committed:
Update README.md

mattsaxon committed:
Update README.md

mattsaxon committed:
Update README.md

mattsaxon committed:
Update README.md

beveradb commented on README.md:
```diff
@@ -1,48 +1,46 @@
+Since mid 2018, the firmware Itead have shipped with Sonoff devices has provided a feature called "LAN Mode" which allows the device to be controlled directly on the local network using a WebSocket or HTTP REST connection on port 8081.
 
-The feature is designed to only be used when there is no connection to the Itead cloud servers (e.g. if your internet connection is down, or their servers are down). As such, it is only enabled when the device is connected to your WiFi network, but unable to reach the Itead servers. You can find out more about the feature in the [Itead FAQ page](https://help.ewelink.cc/hc/en-us/articles/360007134 ...
+Whilst older devices only dropped into LAN moe when the internet was unavailabel, the latest V3 firmware are designed to be used in LAN mode predominantly. Note: the eWeLink app doesn't show this very well and when you turn it to LAN mode, it doesn't report that it connect to V3 devices in this mode, so many users are unaware of this and believe the device to be connected via internet, however th ...
```
A few minor typos / sentence structure tweaks here - happy to tweak myself if merged though.

mattsaxon replied:
I've made some improvements, but please feel free to refine

beveradb commented on switch.py:
```diff
@@ -30,34 +35,56 @@
-        _LOGGER.setLevel(logging.DEBUG)
+    def __init__(self, hass, host, name, icon, device_id, api_key):
+            
+        from pysonofflan3 import SonoffSwitch
```
Ahhh, I see - let's pick this up in the `pysonofflan` [PR thread](https://github.com/beveradb/pysonofflan/pull/79), as this doesn't feel ideal to me.

mattsaxon replied:
I've made some improvements, but please feel free to refine

mattsaxon replied:
Agreed, happy to discuss. It's certainly not ideal.

mattsaxon committed:
Update README.md

mattsaxon committed:
Side by side changes

mattsaxon committed:
Merge branch 'master' of https://github.com/mattsaxon/sonoff-lan-mode-homeassistant

mattsaxon committed:
Update README.md

mattsaxon committed:
Update README.md

mattsaxon committed:
Update README.md

mattsaxon committed:
Update README.md

mattsaxon committed:
Update README.md

mattsaxon committed:
Update README.md

mattsaxon committed:
Update README.md

mattsaxon committed:
Update README.md

mattsaxon committed:
Update README.md

mattsaxon committed:
Update README.md

mattsaxon committed:
Update README.md

mattsaxon committed:
Update sonoff_lan_mode_r3.zip

mattsaxon committed:
Merge branch 'master' of https://github.com/mattsaxon/sonoff-lan-mode-homeassistant

mattsaxon committed:
Update README.md

mattsaxon committed:
Update sonoff_lan_mode_r3.zip

mattsaxon committed:
Merge branch 'master' of https://github.com/mattsaxon/sonoff-lan-mode-homeassistant

mattsaxon committed:
Update README.md

mattsaxon committed:
Update README.md

mattsaxon committed:
Update README.md

mattsaxon committed:
Update sonoff_lan_mode_r3.zip

Deal with long message split across data1, data2, data3 & data4 (for Sonoff POW R2)

mattsaxon committed:
Merge branch 'master' of https://github.com/mattsaxon/sonoff-lan-mode-homeassistant

mattsaxon committed:
Update README.md

mattsaxon committed:
Update sonoff_lan_mode_r3.zip

mattsaxon committed:
Merge branch 'master' of https://github.com/mattsaxon/sonoff-lan-mode-homeassistant

mattsaxon committed:
Update README.md

mattsaxon committed:
Update README.md

mattsaxon committed:
Update sonoff_lan_mode_r3.zip

mattsaxon committed:
Merge branch 'master' of https://github.com/mattsaxon/sonoff-lan-mode-homeassistant

mattsaxon committed:
Draft suport for multi-outlet

Provides support for the first outlet on a multi-outlet device (not yet tested as I don't have such a device!)

mattsaxon committed:
Update sonoff_lan_mode_r3.zip

mattsaxon committed:
Added __init__.py

mattsaxon committed:
Added support for enhnaced plug type devices

mattsaxon committed:
added support for TH

mattsaxon committed:
Update README.md

mattsaxon committed:
Added support for outlet field

mattsaxon committed:
Merge branch 'master' of https://github.com/mattsaxon/sonoff-lan-mode-homeassistant

mattsaxon committed:
fix update issue for multi-outlet

mattsaxon committed:
Improve hack to deal with device removal on some hardware

mattsaxon committed:
Update sonoff_lan_mode_r3.zip

mattsaxon committed:
Update sonoff_lan_mode_r3.zip

mattsaxon committed:
Update sonoff_lan_mode_r3.zip

mattsaxon committed:
Update README.md

mattsaxon committed:
Update sonoff_lan_mode_r3.zip

mattsaxon committed:
Merge branch 'master' of https://github.com/mattsaxon/sonoff-lan-mode-homeassistant

mattsaxon committed:
Update sonoff_lan_mode_r3.zip

mattsaxon committed:
Update sonoff_lan_mode_r3.zip

mattsaxon committed:
Update README.md

mattsaxon committed:
Create requirements_dev.txt

mattsaxon committed:
Create .pyup.yml

mattsaxon committed:
Update README.md

mattsaxon committed:
Update .pyup.yml

mattsaxon committed:
Update README.md

mattsaxon committed:
Update README.md

mattsaxon committed:
Update README.md

mattsaxon committed:
Merge branch 'master' of https://github.com/mattsaxon/sonoff-lan-mode-homeassistant

mattsaxon committed:
Move to use PyPI to distribute the PySonoffLANR3 component

mattsaxon committed:
Update to remove legacy code and support PyUp

mattsaxon committed:
Update README.md

mattsaxon committed:
Update __init__.py

mattsaxon committed:
Update manifest.json

mattsaxon committed:
Update switch.py

mattsaxon committed:
Update README.md

mattsaxon committed:
Fix disconnection issues

mattsaxon committed:
Revert "Fix disconnection issues"

This reverts commit 98c64e2046b6c1c88427a622d0a73146bdcc4c26.

mattsaxon committed:
1.1.3

javicalle committed:
Real logger name

With this PR the logger gets the real package name.
Due to the actual definition I was not able to correctly configure the log traces for this integration.

mattsaxon committed:
Merge pull request #25 from javicalle/patch-1

Real logger name

mattsaxon committed:
Update README.md

mattsaxon committed:
v1.1.4

mattsaxon committed:
Merge branch 'master' of https://github.com/mattsaxon/sonoff-lan-mode-homeassistant

mattsaxon committed:
Support for HACS

mattsaxon committed:
Update README.md

mattsaxon committed:
Update README.md

mattsaxon committed:
Update README.md

mattsaxon committed:
Some linting issues

pyup-bot committed:
Update homeassistant from 0.107.7 to 0.108.9

mattsaxon committed:
Update README.md

mattsaxon committed:
Merge pull request #32 from mattsaxon/pyup-update-homeassistant-0.107.7-to-0.108.9

Update homeassistant to 0.108.9

mattsaxon committed:
Update README.md

pyup-bot committed:
Update homeassistant from 0.108.9 to 0.109.2

mattsaxon committed:
Fix for 0.109

Rename async functions correctly

mattsaxon committed:
Merge pull request #36 from mattsaxon/pyup-update-homeassistant-0.108.9-to-0.109.2

Update homeassistant to 0.109.2

pyup-bot committed:
Update homeassistant from 0.109.2 to 0.109.3

mattsaxon committed:
Update hacs.json

mattsaxon committed:
Merge pull request #38 from mattsaxon/pyup-update-homeassistant-0.109.2-to-0.109.3

Update homeassistant to 0.109.3

hukoeth committed:
Add tested device

I just successfully tested Wifi enabled circuit breaker WDYK ZJSB9-80 Wifi which is also available from Hoch. The device works without any problems out of the box. Firmware is 3.4.0.

pyup-bot committed:
Update homeassistant from 0.109.3 to 0.110.1

hukoeth committed:
Change SwitchDevice to SwitchEnity

mattsaxon committed:
Merge pull request #47 from hukoeth/master

Change SwitchDevice to SwitchEnity

mattsaxon committed:
Merge pull request #46 from mattsaxon/pyup-update-homeassistant-0.109.3-to-0.110.1

Update homeassistant to 0.110.1

mattsaxon committed:
Merge pull request #43 from hukoeth/patch-1

Add tested device

mattsaxon committed:
Update README.md
